### DISCUSSION TITLE
Context menu not showing up on UI Elements

### DISCUSSION THREAD

**Discussion by purohitanuj** [eclipse/windowbuilder#757] [Q&A]
Hi,
In earlier versions of WindowBuilder, I was able to do various tasks on right click on UI Element. For example, I could have placed a button and then right click on it to selection the action of the button, which would open the code editor with that block of new code in place. Another example would be right click on window to assign the "Window Closing" / " Window Closed" event to write a code for that. Now only two options are available. 
Please fix it or let me know how I can fix it.
![Context Menu Issue](https://github.com/eclipse-windowbuilder/windowbuilder/assets/166108048/5d2f813d-1f6c-4614-83e7-1f8bf161ddaf)
See above. Earlier, there were many options were shown if I had right clicked the Button.

![IDE Version](https://github.com/eclipse-windowbuilder/windowbuilder/assets/166108048/7a47061b-34e5-4f66-b7ad-284705f311aa)
![WindowBuilderInfo](https://github.com/eclipse-windowbuilder/windowbuilder/assets/166108048/fd443ba2-b15d-4ad8-b07c-5f871e1559ed)

**Comment by gss0** [comment]
I also encountered this problem. Even after a clean install of eclipse and the plugin, I did not get the results I was hoping for.

I managed to identify the culprit but I do not know how to solve it. Apparently the problem is due to GridBagLayout.

This bug greatly limits productivity and makes it almost impossible to program easily... please fix it as soon as possible or at least make public directives to mitigate the problem.

Thanks

![Screenshot 2024-09-23 174302](https://github.com/user-attachments/assets/235499dc-3cb1-4882-9a71-316cc319e92a)
![Screenshot 2024-09-23 174319](https://github.com/user-attachments/assets/0143245e-5c54-4de1-b52f-c42e6790cc2d)
![Screenshot 2024-09-23 174331](https://github.com/user-attachments/assets/3950fe9c-998e-428d-a129-8a365631fc59)
![Screenshot 2024-09-23 174434](https://github.com/user-attachments/assets/372c7d63-61eb-4174-928a-6436736b387e)
![Screenshot 2024-09-23 175158](https://github.com/user-attachments/assets/149b086d-f963-4139-b5ac-845a288f67a6)

**Reply by ptziegler** [reply]
As mentioned [here](https://github.com/eclipse-windowbuilder/windowbuilder/issues/758#issuecomment-2094801253), the problem only seems to be showing up when using Java 8. So the general recommendation is to update to a supported Java LTS and -if necessary- set a compiler compliance on your project.
It it still happens with a modern Java version, please provide a small reproducer.

Given that I no longer work with such old Java releases, this is not a priority on my side. Meaning that if this is something that urgently needs to be resolved, your best bet is to contribute a fix yourself.
